pytorch_lightning/utilities/data.py

```diff
@@ -336,23 +352,28 @@ def wrapper(obj: DataLoader, *args: Any, **kwargs: Any) -> None:
             if param.name != "self" and param.kind not in (param.VAR_POSITIONAL, param.VAR_KEYWORD)
         ]
 
-        cls = type(obj)
+        set_arg_names = []
         for arg_name, arg_value in chain(zip(param_names, args), kwargs.items()):
-            if hasattr(cls, arg_name) and getattr(cls, arg_name).fset is None:
-                # the class defines a read-only (no setter) property of this name. it's likely that the implementation
-                # will set `self._arg_name = arg_value` in `__init__` which is the attribute returned by the `arg_name`
-                # property so we are fine skipping in that case
-                continue
-            setattr(obj, arg_name, arg_value)
+            if not hasattr(obj, "__" + arg_name):
+                # Set the value privately if it has not been set yet. This achieves two things:
```

Comment:
I understand the benefits achieved, but are they a requirement for the intended changes in this PR?

Reply:
No, they are not a requirement, but I think it's only a matter of time before someone will come with an implementation along the lines of

```python
class MyIncompleteLoader(DataLoader):
    def __init__(self, dataset, batch_size, **kwargs):
        batch_size = min(batch_size - 5, 0) # five last batches are broken
        super().__init__(dataset, batch_size=batch_size, **kwargs)
```

And then will be very surprised, why their loader loads all the batches when they explicitly wanted to not load 5 last batches.

This basically ensures correctness in such a case. Sure, it might be slightly degenerate case, but I don't think it is too far-fetched

Reply:
Added a test case for it